### ISSUE TITLE
Reduce CPU in browser when tab is not visible

### ISSUE THREAD

**Issue by blammit** [victronenergy/gui-v2#327]
Relevant thread with Mathijs:

The main two things:
- disconnect from mqtt
- stop moving ants animation

Note that browser tab not being active is not the same as browser tab not visible.

Which happens because browsers are getting smart: tabs that use a lot of CPU while not visible to the user are getting punished by them to reducing allowed number of CPU cycles (or function calls or however they to it) to a minimum. 

Note: Talk to Alex & co working on VRM Portal and see how they are solving these issues.

If this appears to be impossible from WASM/Qt, then we can do it from html/javascript as well: unload WASM if tab is invisible for 30 seconds (or some other number of seconds) and reload when visible again.

**Comment by chriadam** [comment]
Some work in this area has been done already, see the listed PRs above.  However, those aren't webassembly specific (e.g. in general we need to reduce CPU usage on all screens, regardless of platform), but this task tracks webassembly-specific changes which are needed to reduce tab CPU usage.  I suspect we might need to do things like e.g. stop animations after some time, rather than letting them continue to play.

We will look into this once the other issues are resolved.
Considering it low priority for now.

**Comment by jpetrell** [comment]
What is still missing here? Seems the issue described in the title has been solved already? Nothing has happened here in months, maybe close this one and file remaining issues as separate items.

**Comment by chriadam** [comment]
I agree.  Closing.  One possibility: we could detect when tab is inactive, and use that to pause electron animations, for example.

**Comment by mpvader** [comment]
hi, reopening this to make sure its on the agenda for v1.1.0. I'm not 100% sure if something needs to be implemented here - first need a short discussion with Wiebe, and maybe VRM team to make clear if something needs doing / what needs doing and perhaps sending a signal from VRM to gui-v2 might help as well.

But - this is not a priority for now
